clang/lib/Driver/Driver.cpp

```diff
@@ -5570,6 +5570,21 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
   // When compiling for -fsycl, generate the integration header files and the
   // Unique ID that will be used during the compilation.
   if (Args.hasFlag(options::OPT_fsycl, options::OPT_fno_sycl, false)) {
+    const bool IsSaveTemps = isSaveTempsEnabled();
+    SmallString<128> OutFileDir;
+    if (IsSaveTemps) {
+      if (SaveTemps == SaveTempsObj) {
+        auto *OptO = C.getArgs().getLastArg(options::OPT_o);
+        OutFileDir = (OptO ? OptO->getValues()[0] : "");
+        llvm::sys::path::remove_filename(OutFileDir);
+      }
+      // If in `SaveTempsCwd` mode, or `remove_filename` returned an empty
+      // string use CWD, otherwise append separator.
+      if (OutFileDir.empty())
+        llvm::sys::path::native(OutFileDir = "./");
```

Comment:
I looked through how other temporaries for save-temps are formulated in regards to $CWD, and there is no qualifier like `./`.  The output is just the filename.  Perhaps we should do the same here.

Reply:
I think you are right, removed. Looks like the other use of `./` https://github.com/jchlanda/llvm/blob/jakub/save_temps_llvm_foreach/clang/lib/Driver/ToolChains/SYCL.cpp#L138 will have to stay, because of the way `llvm-foreach` handles the `OutDirectory` option.